app/addons/fauxton/tests/componentsSpec.react.jsx

```diff
@@ -204,5 +207,169 @@ define([
+        msg: 'Success!',
+        time: moment()
+      },
+      info: {
+        notificationId: 2,
+        type: 'info',
+        msg: 'Error!',
+        time: moment()
+      },
+      error: {
+        notificationId: 3,
+        type: 'error',
+        msg: 'Error!',
+        time: moment()
+      }
+    };
+
+    beforeEach(function () {
+      container = document.createElement('div');
+    });
+
+    afterEach(function () {
+      React.unmountComponentAtNode(container);
+    });
+
+    it('shows all notification types when "all" filter applied', function () {
+      var row1 = React.render(
+        <Views.NotificationRow filter="all" item={notifications.success} transitionSpeed={0} />,
+        container
+      );
+      assert.equal($(row1.getDOMNode()).data('visible'), true);
+      React.unmountComponentAtNode(container);
+
+      var row2 = React.render(
+        <Views.NotificationRow filter="all" item={notifications.error} transitionSpeed={0} />,
+        container
+      );
+      assert.equal($(row2.getDOMNode()).data('visible'), true);
+      React.unmountComponentAtNode(container);
+
+      var row3 = React.render(
+        <Views.NotificationRow filter="all" item={notifications.info} transitionSpeed={0} />,
+        container
+      );
+      assert.equal($(row3.getDOMNode()).data('visible'), true);
+      React.unmountComponentAtNode(container);
+    });
+
+    it('hides notification when filter doesn\'t match', function () {
+      var rowEl = React.render(
+        <Views.NotificationRow filter="success" item={notifications.info} transitionSpeed={0} />,
+        container
+      );
+      assert.equal($(rowEl.getDOMNode()).data('visible'), false);
+    });
+
+    it('shows notification when filter exact match', function () {
+      var rowEl = React.render(
+        <Views.NotificationRow filter="info" item={notifications.info} transitionSpeed={0} />,
+        container
+      );
+      assert.equal($(rowEl.getDOMNode()).data('visible'), true);
+    });
+
+  });
+
+
+  describe('NotificationCenterPanel', function () {
+    var container;
+
+    beforeEach(function () {
+      container = document.createElement('div');
+      store.reset();
+    });
+
+    afterEach(function () {
+      React.unmountComponentAtNode(container);
+    });
+
+    it('shows all notifications by default', function () {
+      store.addNotification({ type: 'success', msg: 'Success are okay' });
+      store.addNotification({ type: 'success', msg: 'another success.' });
+      store.addNotification({ type: 'info', msg: 'A single info message' });
+      store.addNotification({ type: 'error', msg: 'Error #1' });
+      store.addNotification({ type: 'error', msg: 'Error #2' });
+      store.addNotification({ type: 'error', msg: 'Error #3' });
+
+      var panelEl = React.render(<Views.NotificationCenterPanel />, container);
+      assert.equal($(panelEl.getDOMNode()).find('.notification-list li[data-visible=true]').length, 6);
```

Comment:
It's just for testing whether it's visible or not. The problem is just timing: any of the nice transition effects we do (bootstrap, velocity) take time to perform so checking visibility doesn't work immediately. This is a workaround. I *could* use a class, but I kinda prefer a data-role because to me it seems more detached from a class which could have styles. I know data-roles could be styled too, but they are less likely to & I like to draw attention to it.